### PR TITLE
Trigger tests with standard set

### LINKED ISSUE #1116: Update how tests are launched via Build Pipelines
2 part feature request:

1) Change updateTests() 
We are generating new test jobs (with a name change) so need to update the way tests are launched (currently based on old job names), see [updateTests()](https://github.com/AdoptOpenJDK/openjdk-build/blob/master/pipelines/build/common/openjdk_build_pipeline.groovy#L96-L130)

2) Re-introduce the ability to add 1 or more levels of testing
We should also add the capability to add extra testing when the build pipeline is for a release, versus a nightly build.  Currently builds launch top-level sanity targets, release builds used to additionally launch extended targets (before the build pipeline refactoring).  

Would like to reintroduce a way to launch both sanity and extended testing for releases as described https://github.com/AdoptOpenJDK/openjdk-tests/wiki/Graduated-Testing-&-Test-Numbers

It would also be good to make it more flexible to schedule different levels of testing, as suggest, to update the nightly build pipelines to run 'the nightly testing' Mon-Fri, and to run 'weekly testing' on the weekends, or something similar.  And finally for release builds, triggering both nightly and weekly testing.

### PR DESCRIPTION
**Pull request by sophia-guo**
Set standard set of top-level test targets run nightly, weekly. Per release combined nightly and weekly targets will run

1. Keep all pipeline jobs runs the same targets as before.
2. Keep job configuration testList as simple as a placeholder
3. Possible key value for testlist

> no test enabled: test: false, or no test as key of configuration.
> standard set -->  test: "default"
> others --> 

```
                test                : [
                        nightly: [customized target lists, could be empty too],
                        weekly : [customized target lists, could be empty too]
                ],
```

Close #1116 

Signed-off-by: Sophia Guo <sophia.gwf@gmail.com>

### CODE REVIEW

**Comment by sophia-guo** [comment]
For now test for x64Linux in  jdk11u_pipeline_config.groovy is updated to "default", which means it will run the full standard daily for nightly build, weekly( Cron triggered) and nightly+weekly for release builds.

Keep others as before, we can go through to see if we need to update for others.

File: pipelines/build/common/build_base_file.groovy
Comment (karianna) [inline pipelines/build/common/build_base_file.groovy:127]: These look like constants to me, should we hoist up to the class level?

File: pipelines/build/common/build_base_file.groovy
Comment (karianna) [inline pipelines/build/common/build_base_file.groovy:138]: Long return statement, prefer splitting it out into individual statements with a simpler return at the end

File: pipelines/build/common/build_base_file.groovy
Comment (karianna) [inline pipelines/build/common/build_base_file.groovy:143]: Is this default else covering both "weekly" and "anything else"?

File: pipelines/build/common/config_regeneration.groovy
Comment (karianna) [inline pipelines/build/common/config_regeneration.groovy:170]: this looks like a constant, should it be extracted and named?

File: pipelines/build/common/build_base_file.groovy
Comment (sophia-guo) [reply pipelines/build/common/build_base_file.groovy]: Yes, updated.

File: pipelines/build/common/build_base_file.groovy
Comment (sophia-guo) [reply pipelines/build/common/build_base_file.groovy]: yes, updated.

File: pipelines/build/common/config_regeneration.groovy
Comment (sophia-guo) [reply pipelines/build/common/config_regeneration.groovy]: updated.

File: pipelines/build/common/build_base_file.groovy
Comment (sophia-guo) [reply pipelines/build/common/build_base_file.groovy:144]: else is cover 'release' job type hence nightly + weekly tests should be run.

**Comment by sophia-guo** [comment]
@smlambert Thanks for reminding. jdk8u updated.

**Comment by smlambert** [comment]
run tests

**Comment by sophia-guo** [comment]
Could someone run a PR build? Thanks.

**Comment by smlambert** [comment]
@karianna - if you are good with @sophia-guo's updates to your review comments, could you trigger pr tests or let me know how I might do so (my run tests comment above did not seem to have an effect).

**Comment by aahlenst** [comment]
run tests

**Comment by sophia-guo** [comment]
The user should be added to White list. Either directly update PR jobs configuration or Admin need to comment 'add to whitelist'.

**Comment by gdams** [comment]
Run tests

**Comment by sophia-guo** [comment]
Run tests

File: pipelines/build/common/config_regeneration.groovy
Comment (karianna) [inline pipelines/build/common/config_regeneration.groovy:172]: whitespace

File: pipelines/jobs/configurations/jdk11u_pipeline_config.groovy
Comment (karianna) [inline pipelines/jobs/configurations/jdk11u_pipeline_config.groovy:7]: I wonder if we need to define the default set in this file instead?  Else the reader has to dig into an unknow source file to figure that out...

File: pipelines/build/common/build_base_file.groovy
Comment (sophia-guo) [reply pipelines/build/common/build_base_file.groovy:148]: Updated by testList.unique() to remove duplicates.

Empty arrays works the same as tests are disabled.

File: pipelines/build/common/config_regeneration.groovy
Comment (sophia-guo) [reply pipelines/build/common/config_regeneration.groovy]: updated

File: pipelines/build/common/build_base_file.groovy
Comment (sophia-guo) [reply pipelines/build/common/build_base_file.groovy:153]: Comment added.

**Review by karianna — APPROVED** [review]

**Comment by karianna** [comment]
run tests

**Comment by sxa** [comment]
@sophia-guo where are we with this? Is it ready to go in once build lockdown is complete?

**Comment by sophia-guo** [comment]
The former PR tests ran good. I haven't got the chance to check the new  pipeline-build-check failure. Will check it and  resolve the conflicts.

**Comment by sophia-guo** [comment]
run tests

**Comment by sophia-guo** [comment]
Run tests

**Comment by sophia-guo** [comment]
It is ready to go in once build lockdown is complete.

**Comment by sophia-guo** [comment]
run tests

**Comment by karianna** [comment]
All good but waiting for OpenJ9 15 release before landing this

**Comment by sophia-guo** [comment]
Can anyone help this? If the PR test build was gone and I have no idea what is the failure. Thanks.

**Comment by karianna** [comment]
run tests

**Review by gdams — APPROVED** [review]

**Pull request merged**
